src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -80,70 +89,184 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
+        # type: (str, str, str) -> str
+        # Reads a *.list file and returns only lines that are functionally required
+        self.composite_logger.log_verbose("[APM] Reading source list for consolidation. [FilePath={0}]".format(file_path))
+
+        std_source_list_content = str()
+        sources_content = self.env_layer.file_system.read_with_retry(file_path)
+        self.composite_logger.log_verbose("[APM][Srclist] Input source list file. [FilePath={0}][Content={1}]".format(file_path, sources_content))
+        sources_content_lines = sources_content.splitlines()
+
+        for line in sources_content_lines:
+            if len(line.strip()) != 0 and not line.strip().startswith("#"):
+                if base_classification == Constants.PackageClassification.SECURITY and "security" not in line:
+                    continue
+                std_source_list_content += "\n" + self.__apply_max_patch_publish_date(sources_content=line, max_patch_publish_date=max_patch_published_date)
+
+        self.composite_logger.log_verbose("[APM][Srclist] Output source list slice. [FilePath={0}][TransformedContent={1}]".format(file_path, std_source_list_content))
+        return std_source_list_content.strip()
+
+    def __read_deb882_style_format(self, file_path, max_patch_published_date, base_classification):
+        # type: (str, str, str) -> str
+        std_source_parts_content = str()
+        stanza = str()
+
+        try:
+            source_parts_content = self.env_layer.file_system.read_with_retry(file_path) + "\n"
+            self.composite_logger.log_verbose("[APM][Deb882] Input source parts file. [FilePath={0}][Content={1}]".format(file_path, source_parts_content))
+            source_parts_content_lines = source_parts_content.splitlines()
+
+            for line in source_parts_content_lines:
+                if line.startswith("#"):    # comments
+                    continue
+
+                if len(line.strip()) == 0:  # stanza separating line
+                    if stanza != str():
+                        if base_classification == str() or (base_classification == Constants.PackageClassification.SECURITY and "security" in stanza):
+                            std_source_parts_content += self.__apply_max_patch_publish_date(sources_content=stanza, max_patch_publish_date=max_patch_published_date) + '\n'
+                        stanza = str()
+                        continue
+
+                stanza += line + '\n'
 
-            self.refresh_repo(sources=self.custom_sources_list)
         except Exception as error:
-            self.composite_logger.log_error("[APM] Error in modifying custom sources list. [Error={0}]".format(repr(error)))
-            return str()    # defaults code to safety
+            self.composite_logger.log_error("[APM][Deb882] Error while reading DEB882-style format. [File={0}][Error={1}]".format(file_path, repr(error)))
+
```

Comment:
line 218 shows up as not covered by tests

Reply:
Covered now